### PR TITLE
fix(App.Modules.Table): validate phonenumber for stupid softphones #196

### PR DESCRIPTION
**Pull request by asdeadasad0do**
Created a pipe for formatting the phonenumber right. Now there aren't any symbols/characters except
numbers and plus

ISSUES CLOSED: #196

### CODE REVIEW

File: Phonebook.Frontend/src/app/modules/rooms/rooms.module.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import { PipesModule } from './../../shared/pipes/pipes.module';
```
Comment (DanielHabenicht) [inline Phonebook.Frontend/src/app/modules/rooms/rooms.module.ts]: please import from `src/`

File: Phonebook.Frontend/src/app/modules/table/components/table/table.component.html [hunk 2]
```diff
@@ -111,8 +115,8 @@
         <button
           mat-button
           actionDrawer
-          [copy]="person.Contacts.Mobile"
-          [tel]="person.Contacts.Mobile"
+          [copy]="person.Contacts.Mobile | telephone"
```
Comment (DanielHabenicht) [inline Phonebook.Frontend/src/app/modules/table/components/table/table.component.html]:
```suggestion
          [copy]="person.Contacts.Mobile"
```

File: Phonebook.Frontend/src/app/modules/table/table.module.ts [hunk 3]
```diff
@@ -1,3 +1,4 @@
+import { PipesModule } from './../../shared/pipes/pipes.module';
```
Comment (DanielHabenicht) [inline Phonebook.Frontend/src/app/modules/table/table.module.ts]: pls import from `src/`

File: Phonebook.Frontend/src/app/shared/components/user/user.module.ts [hunk 4]
```diff
@@ -22,6 +22,7 @@ import { UserSmallCardComponent } from 'src/app/shared/components/user/user-smal
 import { UserTinyCardComponent } from 'src/app/shared/components/user/user-tiny-card/user-tiny-card.component';
 import { ActionDrawerModule } from 'src/app/shared/directives/action-drawer/action-drawer.module';
 import { InteractiveAttributeFieldModule } from 'src/app/shared/interactive-attribute-field/interactive-attribute-field.module';
+import { PipesModule } from '../../pipes/pipes.module';
```
Comment (DanielHabenicht) [inline Phonebook.Frontend/src/app/shared/components/user/user.module.ts]: Please import from `src/`

File: Phonebook.Frontend/src/app/shared/interactive-attribute-field/interactive-attribute-field.module.ts [hunk 5]
```diff
@@ -1,3 +1,4 @@
+import { PipesModule } from './../pipes/pipes.module';
```
Comment (DanielHabenicht) [inline Phonebook.Frontend/src/app/shared/interactive-attribute-field/interactive-attribute-field.module.ts]: Please import from `src/`

File: Phonebook.Frontend/src/app/shared/pipes/tel.pipe.ts [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import { PipeTransform, Pipe } from '@angular/core';
+
+@Pipe({
+  name: 'telephone'
+})
+export class TelephonePipe implements PipeTransform {
```
Comment (DanielHabenicht) [inline Phonebook.Frontend/src/app/shared/pipes/tel.pipe.ts]: Can you add a unit test for this pipe?

**Comment by paule96** [comment]
I have today a excurse with some telephone people. The `+` is important for moving phones. (mobile phones)

So maybe we shouldn't remove this. Please ask @phixom for more information.

**Comment by mschwrdtnr** [comment]
Hey @asdeadasad0do, how it is going? 😸

**Comment by DanielHabenicht** [comment]
@asdeadasad0do ping

**Comment by DanielHabenicht** [comment]
> I have today a excurse with some telephone people. The `+` is important for moving phones. (mobile phones)
> 
> So maybe we shouldn't remove this. Please ask @phixom for more information.

The + is not removed.

File: Phonebook.Frontend/src/app/shared/pipes/telephone.pipe.spec.ts [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import { TelephonePipe } from './telephone.pipe';
+
+describe('TestPipe', () => {
+  it('create an instance', () => {
+    const pipe = new TelephonePipe();
+    expect(pipe).toBeTruthy();
+  });
+
+  it('transforms numer', () => {
```
Comment (mschwrdtnr) [inline Phonebook.Frontend/src/app/shared/pipes/telephone.pipe.spec.ts]: spelling mistake?

Comment (DanielHabenicht) [inline Phonebook.Frontend/src/app/shared/pipes/telephone.pipe.spec.ts]:
```suggestion
  it('transforms number', () => {
```

**Comment by DanielHabenicht** [comment]
The format is from 3 months ago when this PR was opened

**Pull request merged**